ext/fetch/26_fetch.js

```diff
@@ -398,19 +402,22 @@
       request.body = res.body;
     }
     ArrayPrototypePush(request.urlList, () => locationURL.href);
-    return mainFetch(request, true, terminator);
+    return mainFetch(request, true, terminator, isLocalRequest);
   }
 
   /**
    * @param {RequestInfo} input
    * @param {RequestInit} init
    */
   function fetch(input, init = {}) {
+    let isLocalRequest = typeof input === "string" && input.startsWith("/");
```

Comment:
Can you double-check what happens if you pass a `Request` to `fetch` instead of a string? It looks like a `Request` to a local URL wouldn't be handled correctly.